### PR TITLE
Fixes #59: allow read only token generation based on public keys

### LINKED ISSUE #59: Feature: Do not require private key for JWT if not planning to sign tokens?
Right now the JWT module requires a private key for asymetric keys in order to be used, however, if you're only interested in decoding and verifying a key then this is an unnecessary additional piece of security data that needs to be stored in the key store. The piece of code responsible for this is located here:

``` java
X509Certificate certificate = getCertificate(keyStore, alg);
PrivateKey privateKey = getPrivateKey(keyStore, keyStorePassword, alg);
if (certificate != null && privateKey != null) {
  tmp.put(alg, new CryptoSignature(alias.get(alg), certificate, privateKey));
} else {
  log.info(alg + " not available");
}
```

Can the requirement that a private key be present be lifted?

### PR DESCRIPTION
**Pull request by pmlopes**
Fixes #59 

generate tokens using a public key. These tokens cannot be encoded on the host but on the node that has the private key.

### CODE REVIEW

**Comment by pmlopes** [comment]
Still need to add documentation

File: vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/impl/JWT.java [hunk 1]
```diff
@@ -40,6 +42,10 @@
   private final Map<String, Crypto> CRYPTO_MAP;
```
Comment (vietj) [inline vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/impl/JWT.java]: why upper case ?

File: vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/impl/JWTAuthProviderImpl.java [hunk 2]
```diff
@@ -64,7 +64,15 @@ public JWTAuthProviderImpl(Vertx vertx, JsonObject config) {
 
         this.jwt = new JWT(ks, keyStore.getString("password").toCharArray());
       } else {
-        this.jwt = new JWT(null, null);
+        // in the case of not having a key store we will try to load a public key in pem format
+        // this is how keycloak works as an example.
+        String publicKey = config.getString("public-key");
+
+        if (publicKey != null) {
```
Comment (vietj) [inline vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/impl/JWTAuthProviderImpl.java]: given that JWT() calls JWT(null), the test does not seem to make much a difference

**Comment by pmlopes** [comment]
Documentation added and example too

**Pull request merged**